net.go

```diff
@@ -78,3 +81,85 @@ func Listen(network, address string) (net.Listener, error) {
 	// I asked about it here: https://groups.google.com/g/golang-nuts/c/Q1I7Viz9AJc
 	return defaultListenConfig().Listen(context.Background(), network, address)
 }
+
+type Listener struct {
```

Comment:
Let's use exported variables to configure the listener like we do with http servers for instance.